tests/kernel/workq/work_queue/src/main.c

```diff
@@ -730,14 +730,14 @@ static void msg_provider_thread(void *p1, void *p2, void *p3)
 	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
 
-	char msg[MSG_SIZE];
+	char msg[MSG_SIZE] = { 0 };
 
 	k_msgq_put(&triggered_from_msgq_test.msgq, &msg, K_NO_WAIT);
 }
 
 static void triggered_from_msgq_work_handler(struct k_work *work)
 {
-	char msg[MSG_SIZE];
+	char msg[MSG_SIZE] = { 0 };
```

Comment:
No need to initialize this one, right?  k_msgq_get() is going to fill it.  This is a common idiom; there should be no warning condition about passing an uninitialized local array via a non-const pointer, right?

Reply:
Hey you are right! Sorry got carried on and did not realize the two calls were different.